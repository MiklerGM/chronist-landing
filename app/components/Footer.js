import React from 'react';

class Footer extends React.Component {
  render() {
    return (
      <footer className="bg-footer">
        <div>
          <p>Хронист 2017</p>
        </div>
        <ul>
          <li><a href="mailto:contact@chronist.ru">Email </a></li>
          <li><a href='https://vk.com/chronist'><span className="social fa icon-vkontakte" aria-hidden="true"></span></a></li>
        </ul>
      </footer>
    );
  }
}

export default Footer;
