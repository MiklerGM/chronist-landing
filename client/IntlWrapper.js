import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';

import localeDataRU from './locales/ru.json';
// import localeDataEN from './locales/en.json';

class IntlWrapper extends React.Component {
  render() {
    return (
      <IntlProvider
        locale={this.props.locale}
        key={this.props.locale}
        messages={this.props.messages}
      >
        {this.props.children}
      </IntlProvider>
    );
  }
}

IntlWrapper.propTypes = {
  locale: PropTypes.string,
  messages: PropTypes.object
};

IntlWrapper.defaultProps = {
  locale: 'ru',
  messages: localeDataRU
};

export default IntlWrapper;
