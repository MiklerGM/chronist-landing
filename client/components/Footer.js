import React from 'react';
import { Link } from 'react-router-dom';

// import '../styles/footer.less';
if (process.env.WEBPACK) require('../styles/footer.less'); // eslint-disable-line global-require

const Clear = () => (
  <div className='clear' />
);

class FooterDiv extends React.Component {
  render() {
    return (
      <footer id="footer" className='row'><div className='bg-footer'><div className='container'>

        <div className='left-part col-md-6 col-sm-6'>
          <p><span className='icon-copyright' /> 2017. Команда <a href='https://chronist.ru/about' className='social'> Хронист </a></p>
          <p>Художник - <a href='https://www.facebook.com/kotokhira' className='social'>Валерия Журавлёва</a></p>
        </div>
        <div className='right-part col-md-6 col-sm-6'>
          <div className='sns'>
            <a href='mailto:idea@chronist.ru'><span className='social fa icon-envelope-open' aria-hidden='true' /></a>
            <a href='https://vk.com/chronist'><span className='social fa icon-vkontakte' aria-hidden='true' /></a>
          </div>
          <p className='social'><Link to="/contact">Свяжитесь с нами</Link></p>
        </div>
      </div></div><hr /></footer>
    );
  }
}

const Footer = () => (
  <div>
    <Clear />
    <FooterDiv />
  </div>
);

export default Footer;
