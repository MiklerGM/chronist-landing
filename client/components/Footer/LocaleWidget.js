import React from 'react';
import PropTypes from 'prop-types';
import ym from 'react-yandex-metrika';

class LocaleWidget extends React.Component {
  changeToEn() {
    this.props.onChangeLanguage('en');
  }

  changeToRu() {
    this.props.onChangeLanguage('ru');
  }

  toggleLocale() {
    if (this.props.locale === 'en') {
      this.props.onChangeLanguage('ru');
    } else {
      this.props.onChangeLanguage('en');
    }
  }

  render() {
    if (!this.props.min) {
      return (
        <p>
          <button
            onClick={() => this.changeToEn()}
            className='decorless locale--toggle'
            style={(this.props.locale === 'en') ? { color: '#ffffff' } : null}
          >
            English
          </button>
          { '|' }
          <button
            onClick={() => this.changeToRu()}
            className='decorless locale--toggle'
            style={(this.props.locale === 'ru') ? { color: '#ffffff' } : null}
          >
            Русский
          </button>
        </p>
      );
    }
    return (
      <a
        onClick={() => this.toggleLocale()}
        // className='decorless'
      >
        {this.props.locale === 'en' ? 'RU' : 'EN'}
      </a>
    );
  }
}

LocaleWidget.propTypes = {
  locale: PropTypes.string.isRequired,
  onChangeLanguage: PropTypes.func.isRequired
};

export default LocaleWidget;
