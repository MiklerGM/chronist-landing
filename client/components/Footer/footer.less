@import "../../styles/colors.less";

footer {
  width: 100%;
  background-color: @base01;
  position: relative;
  color: @base04;

  & p {
    color: @white;
    font-size: 14px;
    font-weight: 700;
    width: 20%;
    display: inline-block;

    @media screen and (max-width: 768px) {
      width: 100%;
    }
  }

  & a {
    color: @base04;
    box-shadow: inset 0 -0.5px @white;
  }

  & a:hover,
  button:hover {
    color: @base0E;
    text-decoration: none;
  }

  & .fa {
    font-size: 14px;
    line-height: 1em;
  }

  & .sns {
    padding-bottom: 7px;
  }
}

.clear {
  clear: both;
}
