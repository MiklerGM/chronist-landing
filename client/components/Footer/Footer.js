import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';

import LocaleWidget from './LocaleWidget';
import './footer.less';

const Footer = ({ onChangeLanguage, locale }) => (
  <footer className='layer-4'>
    <div className='page--content'>
      <p>
        <Link to='/privacy'>
          <span><FormattedMessage id='footer.privacy' /></span>
        </Link>
      </p>
      <p>
        <Link to='/legal'>
          <span><FormattedMessage id='footer.tos' /></span>
        </Link>
      </p>
      <p className='social'>
        <Link to="/contact">
          <span><FormattedMessage id='footer.contact' /></span>
        </Link>
      </p>
      <p className='social'>
        <Link to="/license">
          <span><FormattedMessage id='footer.license' /></span>
        </Link>
      </p>
      <LocaleWidget onChangeLanguage={onChangeLanguage} locale={locale} />
    </div>
  </footer>
);

Footer.propTypes = {
  onChangeLanguage: PropTypes.func.isRequired,
  locale: PropTypes.string.isRequired
};

export default Footer;
