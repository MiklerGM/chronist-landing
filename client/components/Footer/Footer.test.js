import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import IntlWrapper from '../../IntlWrapper';
import Footer from './Footer';

const TestComponent = () => (
  <IntlWrapper>
    <Router>
      <Footer onChangeLanguage={(e) => console.log('dump func', e)} locale='en' />
    </Router>
  </IntlWrapper>
);

// it must render
it('Footer render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

it('Footer render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

// Snapshot tests
it('Footer Snapshot', () => {
  const tree = renderer
    .create(<TestComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
