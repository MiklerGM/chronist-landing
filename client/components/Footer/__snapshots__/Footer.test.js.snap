// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot 1`] = `
<footer
  className="layer-4"
>
  <div
    className="page--content"
  >
    <p>
      <a
        href="/privacy"
        onClick={[Function]}
      >
        <span>
          Политика конфиденциальности
        </span>
      </a>
    </p>
    <p>
      <a
        href="/legal"
        onClick={[Function]}
      >
        <span>
           Правила пользования сервисом
        </span>
      </a>
    </p>
    <p
      className="social"
    >
      <a
        href="/contact"
        onClick={[Function]}
      >
        <span>
          Свяжитесь с нами
        </span>
      </a>
    </p>
    <p
      className="social"
    >
      <a
        href="/license"
        onClick={[Function]}
      >
        <span>
          Лицензия
        </span>
      </a>
    </p>
    <p>
      <button
        className="locale--toggle active"
        onClick={[Function]}
      >
        English
      </button>
       | 
      <button
        className="locale--toggle "
        onClick={[Function]}
      >
        Русский
      </button>
    </p>
  </div>
</footer>
`;
